queries.sql

```diff
@@ -15,3 +15,33 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* Query and update table */
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELETE_DOB;
+UPDATE animals SET weight_kg = (weight_kg * -1);
+ROLLBACK TO DELETE_DOB;
+UPDATE animals SET weight_kg = (weight_kg * -1) WHERE weight_kg < 0; 
+COMMIT TRANSACTION;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT neutered , MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY neutered;
+SELECT neutered, AVG(escape_attempts) FROM animals WHERE EXTRACT(year FROM date_of_birth) BETWEEN 1990 AND 2000 GROUP BY neutered;
```

Comment:
* I think you can select animals grouped by species for this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?)  for example this query targets a particular animal type from the animal table where the date of birth is between 1997 to 2021 `SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;  WHERE  date_of_birth >= '1997/01/01' AND date_of_birth <= '2021/31/12'`
* This is because the species column is used to specify the animal type 
* For this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?) you should add `SELECT species and then group by species` to your answer this way your query is targeting a specific animal type whose birthdate falls between `1990 to 2000`

Reply:
Done